### ISSUE TITLE
isis2std cannot write JPEG images

### ISSUE THREAD

**Issue by rvwagner** [DOI-USGS/ISIS3#3868]
**ISIS version(s) affected**: 3.6.0 - 4.0.1

**Description**  
isis2std cannot produce jpeg images, instead it says `**PROGRAMMER ERROR** Cannot export image as format [jpeg] in ImageExporter.cpp at 400.`.  Tested with ISIS version 3.10.1 on MacOS (freshly installed last month) and a whole bunch of versions on Linux. The bug seems to have been introduced in Version 3.6.0; Version 3.5.2 works.

**How to reproduce**  

    makecube to=t.cub samples=100 lines=100 bands=1
    isis2std from=t.cub to=t.jpg format=JPEG

Result:

    **PROGRAMMER ERROR** Cannot export image as format [jpeg] in ImageExporter.cpp at 400.

**Comment by thareUSGS** [comment]
Hmm - there must be a env. conflict or something as I did your simple test successfully on a couple versions (Linux only though). To test again, I guess I would start with a fresh conda env. and try a re-install... 
```
Initialized 3.9.1
(ISIS 3.9.1) bash-4.2$ makecube to=t.cub samples=100 lines=100 bands=1
(ISIS 3.9.1) bash-4.2$ isis2std from=t.cub to=t.jpg format=JPEG
Group = Results
  OutputFileName = t.jpg
  InputMinimum   = 0.0
  InputMaximum   = 1.0
End_Group
```

**Comment by rvwagner** [comment]
Sure, I can try a fresh install.  For reference, this is MacOS 10.14.6.  I can't fiddle with the Linux systems involved- that's the LROC ISIS installation, and it's possible that there's something weird going on there, as all the GUI programs have been slightly off since version 3.6.0 in a number of ways that aren't relevant to this testing on my personal computer.

In summary, it's still broken with a fresh install of ISIS 4.1.0.

Input:

    $ conda create -n isis_jpeg_test python=3.6

Output:

    ## Package Plan ##
    
     environment location: /Users/rwagner/opt/anaconda3/envs/isis_jpeg_test
    
     added / updated specs:
       - python=3.6
    
    
    The following packages will be downloaded:
    
       package                    |            build
       ---------------------------|-----------------
       libffi-3.2.1               |    h4a8c4bd_1007          42 KB  conda-forge
       openssl-1.1.1g             |       h0b31af3_0         1.9 MB  conda-forge
       pip-20.1                   |     pyh9f0ad1d_0         1.1 MB  conda-forge
       python-3.6.10              |h4334963_1011_cpython        20.2 MB  conda-forge
       readline-8.0               |       hcfe32e1_0         415 KB  conda-forge
       setuptools-46.2.0          |   py36h9f0ad1d_0         653 KB  conda-forge
       sqlite-3.30.1              |       h93121df_0         2.5 MB  conda-forge
       wheel-0.34.2               |             py_1          24 KB  conda-forge
       xz-5.2.5                   |       h0b31af3_0         266 KB  conda-forge
       zlib-1.2.11                |    h0b31af3_1006         101 KB  conda-forge
       ------------------------------------------------------------
                                              Total:        27.1 MB
    
    The following NEW packages will be INSTALLED:
    
     ca-certificates    conda-forge/osx-64::ca-certificates-2020.4.5.1-hecc5488_0
     certifi            conda-forge/osx-64::certifi-2020.4.5.1-py36h9f0ad1d_0
     libcxx             conda-forge/osx-64::libcxx-10.0.0-h1af66ff_2
     libffi             conda-forge/osx-64::libffi-3.2.1-h4a8c4bd_1007
     ncurses            conda-forge/osx-64::ncurses-6.1-h0a44026_1002
     openssl            conda-forge/osx-64::openssl-1.1.1g-h0b31af3_0
     pip                conda-forge/noarch::pip-20.1-pyh9f0ad1d_0
     python             conda-forge/osx-64::python-3.6.10-h4334963_1011_cpython
     python_abi         conda-forge/osx-64::python_abi-3.6-1_cp36m
     readline           conda-forge/osx-64::readline-8.0-hcfe32e1_0
     setuptools         conda-forge/osx-64::setuptools-46.2.0-py36h9f0ad1d_0
     sqlite             conda-forge/osx-64::sqlite-3.30.1-h93121df_0
     tk                 conda-forge/osx-64::tk-8.6.10-hbbe82c9_0
     wheel              conda-forge/noarch::wheel-0.34.2-py_1
     xz                 conda-forge/osx-64::xz-5.2.5-h0b31af3_0
     zlib               conda-forge/osx-64::zlib-1.2.11-h0b31af3_1006
    
    
    Proceed ([y]/n)? y
    ...

Input:

    $ conda activate isis_jpeg_test
    $ conda config --env --add channels conda-forge
    $ conda config --env --add channels usgs-astrogeology
    $ conda config --show channels

Output:

    channels:
      - usgs-astrogeology
      - conda-forge
      - defaults

Input:

    $ conda install -c usgs-astrogeology isis

Output:

    ## Package Plan ##
    
     environment location: /Users/rwagner/opt/anaconda3/envs/isis_jpeg_test
    
     added / updated specs:
       - isis
    
    
    The following packages will be downloaded:
    
       package                    |            build
       ---------------------------|-----------------
       ale-0.7.2                  |   py36h855b5bd_1         169 KB  conda-forge
       armadillo-9.200.4          |    hcb88e47_1205         536 KB  conda-forge
       arpack-3.6.3               |    h06d5271_1005         185 KB  conda-forge
       eigen-3.3.7                |    ha1b3eb9_1001         967 KB  conda-forge
       embree-2.16.0              |       h6834224_0        51.6 MB  usgs-astrogeology
       isis-4.1.0                 |                0        13.9 MB  usgs-astrogeology
       libpq-10.6                 |    hbe1e24e_1000         2.5 MB  conda-forge
       libssh2-1.8.0              |    hf30b1f0_1003         221 KB  conda-forge
       mesalib-18.0.0             |       hb6cfc13_1        12.6 MB  conda-forge
       metis-5.1.0                |    h4a8c4bd_1005         4.1 MB  conda-forge
       nanoflann-1.3.1            |                0          18 KB  conda-forge
       pcre-8.44                  |       h4a8c4bd_0         222 KB  conda-forge
       postgresql-10.6            |    ha1bbaa7_1000         4.1 MB  conda-forge
       tzcode-2020a               |       h0b31af3_0         405 KB  conda-forge
       xorg-libx11-1.6.9          |       h0b31af3_0         873 KB  conda-forge
       ------------------------------------------------------------
                                              Total:        92.3 MB
    
    The following NEW packages will be INSTALLED:
    
     ale                conda-forge/osx-64::ale-0.7.2-py36h855b5bd_1
     armadillo          conda-forge/osx-64::armadillo-9.200.4-hcb88e47_1205
     arpack             conda-forge/osx-64::arpack-3.6.3-h06d5271_1005
     blas               conda-forge/osx-64::blas-1.1-openblas
     boost              conda-forge/osx-64::boost-1.68.0-py36h9888f84_1001
     boost-cpp          conda-forge/osx-64::boost-cpp-1.68.0-h6f8c590_1000
     bullet             usgs-astrogeology/osx-64::bullet-2.86.1-0
     bz2file            conda-forge/noarch::bz2file-0.98-py_0
     bzip2              conda-forge/osx-64::bzip2-1.0.8-h0b31af3_2
     cairo              conda-forge/osx-64::cairo-1.14.6-4
     cspice             conda-forge/osx-64::cspice-66-h0b31af3_1007
     curl               conda-forge/osx-64::curl-7.64.0-heae2a1f_0
     decorator          conda-forge/noarch::decorator-4.4.2-py_0
     eigen              conda-forge/osx-64::eigen-3.3.7-ha1b3eb9_1001
     embree             usgs-astrogeology/osx-64::embree-2.16.0-h6834224_0
     expat              conda-forge/osx-64::expat-2.2.9-h4a8c4bd_2
     ffmpeg             conda-forge/osx-64::ffmpeg-3.4.1-0
     flann              conda-forge/osx-64::flann-1.9.1-0
     fontconfig         conda-forge/osx-64::fontconfig-2.12.1-4
     freetype           conda-forge/osx-64::freetype-2.7-1
     geos               conda-forge/osx-64::geos-3.7.3-h4a8c4bd_0
     geotiff            conda-forge/osx-64::geotiff-1.4.2-hb54a4aa_1
     gettext            conda-forge/osx-64::gettext-0.19.8.1-h46ab8bc_1002
     giflib             conda-forge/osx-64::giflib-5.2.1-h0b31af3_2
     glib               conda-forge/osx-64::glib-2.51.4-0
     gmm                usgs-astrogeology/osx-64::gmm-5.0-h6aef312_0
     gmp                conda-forge/osx-64::gmp-6.2.0-h4a8c4bd_2
     gsl                conda-forge/osx-64::gsl-2.6-ha2d443c_0
     harfbuzz           conda-forge/osx-64::harfbuzz-1.3.4-2
     hdf5               conda-forge/osx-64::hdf5-1.8.18-3
     icu                conda-forge/osx-64::icu-58.2-h0a44026_1000
     isis               usgs-astrogeology/osx-64::isis-4.1.0-0
     jama               usgs-astrogeology/osx-64::jama-125-0
     jasper             conda-forge/osx-64::jasper-1.900.1-4
     jpeg               pkgs/main/osx-64::jpeg-9b-he5867d9_2
     kakadu             usgs-astrogeology/osx-64::kakadu-1-0
     krb5               conda-forge/osx-64::krb5-1.16.3-h24a3359_1000
     libblas            conda-forge/osx-64::libblas-3.8.0-11_openblas
     libcblas           conda-forge/osx-64::libcblas-3.8.0-11_openblas
     libcurl            conda-forge/osx-64::libcurl-7.64.0-h76de61e_0
     libedit            conda-forge/osx-64::libedit-3.1.20170329-hcfe32e1_1001
     libgcc             conda-forge/osx-64::libgcc-4.8.5-hdbeacc1_10
     libgfortran        conda-forge/osx-64::libgfortran-3.0.1-0
     libiconv           conda-forge/osx-64::libiconv-1.15-h0b31af3_1006
     liblapack          conda-forge/osx-64::liblapack-3.8.0-11_openblas
     libopenblas        pkgs/main/osx-64::libopenblas-0.3.6-hdc02c5d_2
     libpng             conda-forge/osx-64::libpng-1.6.37-hbbe82c9_1
     libpq              conda-forge/osx-64::libpq-10.6-hbe1e24e_1000
     libprotobuf        conda-forge/osx-64::libprotobuf-3.9.1-hfbae3c0_0
     libssh2            conda-forge/osx-64::libssh2-1.8.0-hf30b1f0_1003
     libtiff            conda-forge/osx-64::libtiff-4.0.9-he6b73bb_1
     libwebp            conda-forge/osx-64::libwebp-0.5.2-7
     libxcb             conda-forge/osx-64::libxcb-1.13-h1de35cc_1002
     libxml2            conda-forge/osx-64::libxml2-2.9.9-hd80cff7_2
     mesalib            conda-forge/osx-64::mesalib-18.0.0-hb6cfc13_1
     metis              conda-forge/osx-64::metis-5.1.0-h4a8c4bd_1005
     mysql              conda-forge/osx-64::mysql-5.7.20-h0a44026_1001
     mysql-connector-c  conda-forge/osx-64::mysql-connector-c-6.1.11-had4e77e_1002
     nanoflann          conda-forge/osx-64::nanoflann-1.3.1-0
     networkx           conda-forge/noarch::networkx-2.4-py_1
     nlohmann_json      conda-forge/osx-64::nlohmann_json-3.7.3-h4a8c4bd_1
     nn                 conda-forge/osx-64::nn-1.86.0-h1de35cc_2003
     numpy              conda-forge/osx-64::numpy-1.13.3-py36_blas_openblas_200
     openblas           conda-forge/osx-64::openblas-0.2.19-2
     opencv             conda-forge/osx-64::opencv-3.2.0-np113py36_blas_openblas_203
     pcl                conda-forge/osx-64::pcl-1.9.1-hdd77166_1002
     pcre               conda-forge/osx-64::pcre-8.44-h4a8c4bd_0
     pixman             conda-forge/osx-64::pixman-0.34.0-h1de35cc_1003
     postgresql         conda-forge/osx-64::postgresql-10.6-ha1bbaa7_1000
     proj4              conda-forge/osx-64::proj4-4.9.3-h01d97ff_9
     protobuf           conda-forge/osx-64::protobuf-3.9.1-py36h6de7cb9_0
     pthread-stubs      conda-forge/osx-64::pthread-stubs-0.4-h1de35cc_1001
     pvl                conda-forge/noarch::pvl-0.3.0-py_1
     python-dateutil    conda-forge/noarch::python-dateutil-2.8.1-py_0
     pytz               conda-forge/noarch::pytz-2020.1-pyh9f0ad1d_0
     pyyaml             conda-forge/osx-64::pyyaml-5.3.1-py36h37b9a7d_0
     qhull              usgs-astrogeology/osx-64::qhull-7.2.0-0
     qt                 usgs-astrogeology/osx-64::qt-5.9.6-0
     qwt                usgs-astrogeology/osx-64::qwt-6.1.3-0
     scipy              conda-forge/osx-64::scipy-1.2.1-py36hbd7caa9_1
     six                conda-forge/noarch::six-1.14.0-py_1
     spiceypy           conda-forge/noarch::spiceypy-2.3.2-py_0
     suitesparse        conda-forge/osx-64::suitesparse-5.7.2-h0e59142_0
     superlu            conda-forge/osx-64::superlu-5.2.1-hbced767_1205
     tbb                conda-forge/osx-64::tbb-2019.9-ha1b3eb9_1
     tnt                usgs-astrogeology/osx-64::tnt-126-0
     tzcode             conda-forge/osx-64::tzcode-2020a-h0b31af3_0
     x264               conda-forge/osx-64::x264-20131218-0
     xerces-c           usgs-astrogeology/osx-64::xerces-c-3.1.4-h10f7eb2_0
     xorg-fixesproto    conda-forge/osx-64::xorg-fixesproto-5.0-h1de35cc_1002
     xorg-inputproto    conda-forge/osx-64::xorg-inputproto-2.3.2-h1de35cc_1002
     xorg-kbproto       conda-forge/osx-64::xorg-kbproto-1.0.7-h1de35cc_1002
     xorg-libice        conda-forge/osx-64::xorg-libice-1.0.10-h01d97ff_0
     xorg-libsm         conda-forge/osx-64::xorg-libsm-1.2.3-h01d97ff_1000
     xorg-libx11        conda-forge/osx-64::xorg-libx11-1.6.9-h0b31af3_0
     xorg-libxau        conda-forge/osx-64::xorg-libxau-1.0.9-h1de35cc_0
     xorg-libxdmcp      conda-forge/osx-64::xorg-libxdmcp-1.1.3-h01d97ff_0
     xorg-libxext       conda-forge/osx-64::xorg-libxext-1.3.4-h01d97ff_0
     xorg-libxfixes     conda-forge/osx-64::xorg-libxfixes-5.0.3-h01d97ff_1004
     xorg-libxi         conda-forge/osx-64::xorg-libxi-1.7.10-h01d97ff_0
     xorg-xextproto     conda-forge/osx-64::xorg-xextproto-7.3.0-h1de35cc_1002
     xorg-xproto        conda-forge/osx-64::xorg-xproto-7.0.31-h1de35cc_1007
     yaml               conda-forge/osx-64::yaml-0.2.4-h0b31af3_0
    
    The following packages will be DOWNGRADED:
    
     openssl                                 1.1.1g-h0b31af3_0 --> 1.0.2u-h0b31af3_0
     python                       3.6.10-h4334963_1011_cpython --> 3.6.7-h4a56312_1002
     readline                                   8.0-hcfe32e1_0 --> 7.0-hcfe32e1_1001
     sqlite                                  3.30.1-h93121df_0 --> 3.28.0-h9721f7c_0
    
    Proceed ([y]/n)? y
    ...

Input:

    $ python $CONDA_PREFIX/scripts/isis3VarInit.py --data-dir=/Volumes/Disk_SSD/ISIS_DATA2
    $ pushd $ISISDATA
    $ rsync -azv --delete --partial isisdist.astrogeology.usgs.gov::isis3data/data/base .
    $ popd
    $ mv ~/.Isis ~/.BKP_Isis
    $ makecube to=t.cub samples=100 lines=100 bands=1
    $ isis2std from=t.cub to=t.jpg format=JPEG

Output:

    **PROGRAMMER ERROR** Cannot export image as format [jpeg].

(Note: The "ISIS3 Data Area" section of the installation instructions should probably call attention to the change from $ISIS3DATA to $ISISDATA between 3.X and 4.X.  It still lists $ISIS3DATA.)

**Comment by jessemapel** [comment]
I wonder if this is related to #3352. There's a chance this goes away with moving to the conda-forge QT build.

**Comment by jlaura** [comment]
@rvwagner Is this still an issue using the newest release? Thanks!

**Comment by rvwagner** [comment]
I just updated to ISIS 4.2.0, and am getting the same failure when trying to make a jpeg with isis2std.

Here's what the update did, if that's relevant:
```
## Package Plan ##

  environment location: /Users/rwagner/opt/anaconda3/envs/isis_jpeg_test

  added / updated specs:
    - isis


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ale-0.8.4                  |   py36hadcdec8_0         198 KB  conda-forge
    bzip2-1.0.8                |       haf1e3a3_2         154 KB  conda-forge
    cspice-66                  |    h0b31af3_1010         9.7 MB  conda-forge
    expat-2.2.9                |       hb1e8313_2         127 KB  conda-forge
    isis-4.2.0                 |                0        13.9 MB  usgs-astrogeology
    libblas-3.8.0              |      17_openblas          11 KB  conda-forge
    libcblas-3.8.0             |      17_openblas          11 KB  conda-forge
    libcxx-10.0.1              |       h5f48129_0         1.0 MB  conda-forge
    libedit-3.1.20191231       |       hed1e85f_2         103 KB  conda-forge
    libffi-3.2.1               |    hb1e8313_1007          42 KB  conda-forge
    liblapack-3.8.0            |      17_openblas          11 KB  conda-forge
    libopenblas-0.3.10         |       h0794777_0         4.5 MB
    libpng-1.6.37              |       hb0a8c7a_2         319 KB  conda-forge
    nlohmann_json-3.9.1        |       hb1e8313_0         120 KB  conda-forge
    pip-20.2.2                 |             py_0         1.1 MB  conda-forge
    setuptools-49.6.0          |   py36h9f0ad1d_0         924 KB  conda-forge
    six-1.15.0                 |     pyh9f0ad1d_0          14 KB  conda-forge
    tk-8.6.10                  |       hb0a8c7a_0         3.3 MB  conda-forge
    wheel-0.35.1               |     pyh9f0ad1d_0          29 KB  conda-forge
    xorg-libx11-1.6.11         |       haf1e3a3_0         868 KB  conda-forge
    xz-5.2.5                   |       haf1e3a3_1         228 KB  conda-forge
    yaml-0.2.5                 |       haf1e3a3_0          81 KB  conda-forge
    zlib-1.2.11                |             1007         102 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        36.7 MB

The following packages will be UPDATED:

  ale                                  0.7.2-py36h855b5bd_1 --> 0.8.4-py36hadcdec8_0
  ca-certificates                     2020.4.5.1-hecc5488_0 --> 2020.6.20-hecda079_0
  certifi                         2020.4.5.1-py36h9f0ad1d_0 --> 2020.6.20-py36h9f0ad1d_0
  cspice                                   66-h0b31af3_1007 --> 66-h0b31af3_1010
  isis                                              4.1.0-0 --> 4.2.0-0
  libblas                                 3.8.0-11_openblas --> 3.8.0-17_openblas
  libcblas                                3.8.0-11_openblas --> 3.8.0-17_openblas
  libcxx                                  10.0.0-h1af66ff_2 --> 10.0.1-h5f48129_0
  libedit                        3.1.20170329-hcfe32e1_1001 --> 3.1.20191231-hed1e85f_2
  liblapack                               3.8.0-11_openblas --> 3.8.0-17_openblas
  libopenblas                              0.3.6-hdc02c5d_2 --> 0.3.10-h0794777_0
  libpng                                  1.6.37-hbbe82c9_1 --> 1.6.37-hb0a8c7a_2
  nlohmann_json                            3.7.3-h4a8c4bd_1 --> 3.9.1-hb1e8313_0
  pip                                     20.1-pyh9f0ad1d_0 --> 20.2.2-py_0
  setuptools                          46.2.0-py36h9f0ad1d_0 --> 49.6.0-py36h9f0ad1d_0
  six                                           1.14.0-py_1 --> 1.15.0-pyh9f0ad1d_0
  wheel                                         0.34.2-py_1 --> 0.35.1-pyh9f0ad1d_0
  xorg-libx11                              1.6.9-h0b31af3_0 --> 1.6.11-haf1e3a3_0
  xz                                       5.2.5-h0b31af3_0 --> 5.2.5-haf1e3a3_1
  yaml                                     0.2.4-h0b31af3_0 --> 0.2.5-haf1e3a3_0
  zlib                                 1.2.11-h0b31af3_1006 --> 1.2.11-1007

The following packages will be DOWNGRADED:

  bzip2                                    1.0.8-h0b31af3_2 --> 1.0.8-haf1e3a3_2
  expat                                    2.2.9-h4a8c4bd_2 --> 2.2.9-hb1e8313_2
  libffi                                3.2.1-h4a8c4bd_1007 --> 3.2.1-hb1e8313_1007
  tk                                      8.6.10-hbbe82c9_0 --> 8.6.10-hb0a8c7a_0
```

**Comment by jlaura** [comment]
@rvwagner Thanks! I got this added to our support sprint prioritization: https://astrodiscuss.usgs.gov/t/software-support-q4-mission-and-product-support-sprints/460

**Comment by scsides** [comment]
This is repeatable on Mac and Linux systems that do not have a jpeg library installed in the OS (i.e., not in the conda environment). This issue is caused by the usgs-astrogeology conda channel version of the Qt libraries being built with system installed jpeg libraries and the running on systems where they are not.

Possible solution: Rebuild the Qt libraries with qt:jpeg instead of system:jpeg
Better solution: Finalize the Conda Forge Qt builds with the qt:jpeg config

**Comment by krlberry** [comment]
Hi @rvwagner, 
I updated the usgs-astrogeology build of qt to use qt's built-in JPEG library, which looks like it may have been the source of the issue. Would you be able to test this on your side with a new installation of ISIS 4.4.0 and let us know if it solves the issue? (The new build of qt should get used automatically.)

**Comment by rvwagner** [comment]
Looks like it's working with version 4.4.0.  Installed via `conda install -c usgs-astrogeology isis=4.4.0` (and confirmed that the version file said 4.4.0- my first attempt at a clean install following the github install instructions gave me 4.2.0, and I don't know if that's expected), tried my test case from the first post, and I got a working JPEG.  I also tried the same test case on an older version of ISIS on the same computer and that didn't work, so it looks like 4.4.0 is what fixed it.
